crates/storage/src/codec/primitive.rs

```diff
@@ -0,0 +1,100 @@
+//! The module contains the implementation of the `Postcard` codec.
+//! The codec is used for types that can be represented by an array.
+//! It includes all primitive types and types that are arrays inside
+//! or could be represented by arrays.
+
+use crate::codec::{
+    Decode,
+    Encode,
+};
+use fuel_core_types::{
+    blockchain::primitives::DaBlockHeight,
+    fuel_tx::{
+        TxId,
+        UtxoId,
+    },
+    fuel_types::BlockHeight,
+};
+
+/// The codec is used for types that can be represented by an array.
+/// The `SIZE` const specifies the size of the array used to represent the type.
+pub struct Primitive<const SIZE: usize>;
+
+macro_rules! impl_encode {
+    ($($ty:ty, $size:expr),*) => {
+        $(
+            impl Encode<$ty> for Primitive<{ $size }> {
+                type Encoder<'a> = [u8; { $size }];
+
+                fn encode(t: &$ty) -> Self::Encoder<'_> {
+                    t.to_be_bytes()
+                }
+            }
+        )*
+    };
+}
+macro_rules! impl_decode {
+    ($($ty:ty, $size:expr),*) => {
+        $(
+            impl Decode<$ty> for Primitive<{ $size }> {
+                fn decode(bytes: &[u8]) -> anyhow::Result<$ty> {
```

Comment:
Maybe we should use dedicated encoding errors instead of anyhow?

Reply:
Maybe the problem is that  I don't know what the enum will look like=D We have only two errors right now: postcard and `TryFromSliceError`.

 The idea is to make the `Decode` trait as generic as possible and it means something like:

```rust
enum DecodeError {
    TryFromSliceError(TryFromSliceError),
    Other(anyhow::Error)
}
```

And I decided to go simply with `anyhow::Error`.